### PR TITLE
[pybind11] Modernize port + better python dependency handling

### PR DESCRIPTION
**Pull request by theblackunknown**
Modernize port

Avoid using system python, instead leverage python3 port to keep everything consistent
I did not tackled the hardcoded python version, another PR is welcome to tackle this

NOTE: This PR relies on #9978 , so it needs to be rebased/udpated once it will be merged

What does your PR fix?
No logged issue, but the problem arose when trying to use this port to build Python extensions using pybind11 on Linux : 
- Dependency on system Python is not always good (e.g. missing shared libraries)
- Shared library are needed, even when we are using VCPKG_LIBRARY_LINKAGE as static

Which triplets are supported/not supported? Have you updated the CI baseline?
No CI update, works at least with windows & linux

Does your PR follow the maintainer guide?
It should !

### CODE REVIEW

File: ports/pybind11/portfile.cmake [hunk 1]
```diff
@@ -9,29 +9,55 @@ vcpkg_from_github(
 )
```
Comment (LilyWangL) [inline ports/pybind11/portfile.cmake]: `include(vcpkg_common_functions)` is deprecated. Please remove it.

**Review by LilyWangL — CHANGES_REQUESTED** [review]

**Comment by LilyWangL** [comment]
@theblackunknown Thanks for your PR. This PR build failed on Linux with this error:
```
CMake Error at tools/FindPythonLibsNew.cmake:96 (message):
  Python config failure:

  Traceback (most recent call last):

    File "<string>", line 1, in <module>

  ImportError: cannot import name 'sysconfig'

Call Stack (most recent call first):
  /ci/myagent/_work/3/s/scripts/buildsystems/vcpkg.cmake:285 (_find_package)
  tools/pybind11Tools.cmake:16 (find_package)
  CMakeLists.txt:33 (include)
```
And this PR build failed on Windows with this error:
```
CMake Error at tools/FindPythonLibsNew.cmake:161 (message):
  Python libraries not found
Call Stack (most recent call first):
  C:/vsts/_work/3/s/scripts/buildsystems/vcpkg.cmake:285 (_find_package)
  tools/pybind11Tools.cmake:16 (find_package)
  CMakeLists.txt:33 (include)
```
Could you please cotinue to fix this PR?

**Comment by theblackunknown** [comment]
Hi @LilyWangL, this PR is in draft it needs #9978 to be approved and merged first
So I would not expect it to work in the meantime

I'll update this one and ping you once it is merged

**Comment by LilyWangL** [comment]
> Hi @LilyWangL, this PR is in draft because it needs #9978 to be approved and merged first
> So I would not expect it to work in the meantime
> 
> I'll update this one and ping you once it is merged

OK.

**Comment by LilyWangL** [comment]
@theblackunknown This port build failed on Linux and Windows. You can click **Details** to check the CI testing error log.

**Comment by LilyWangL** [comment]
Pinging @theblackunknown for response. Is work still being done for this PR?

**Comment by theblackunknown** [comment]
@LilyWangL yes I am still working on it however it is a slow progress with the recent events

**Comment by LilyWangL** [comment]
Any progress?

**Comment by theblackunknown** [comment]
Hi @LilyWangL, I hardly find time those days to progress on this PR
If you want to make a bit of cleaning on this repo PRs, I am fine to close it and re-submit it later
What do you think ?

**Comment by LilyWangL** [comment]
> Hi @LilyWangL, I hardly find time those days to progress on this PR
> If you want to make a bit of cleaning on this repo PRs, I am fine to close it and re-submit it later
> What do you think ?

Thanks for your reply. I am closing this PR for now, if you have time to handle this PR please reopen it.

**Pull request closed without merging**